js/source/entries/qualitycontrol.js

```diff
@@ -0,0 +1,493 @@
+import '../legacy/jquery.js';
+import '../legacy/d3/d3v4Min.js';
+
+var version = '0.01';
+
+export function init(main_div) {
+    if (!(main_div instanceof HTMLElement)) {
+        main_div = document.getElementById(main_div.startsWith("#") ? main_div.slice(1) : main_div);
+    }
+
+    var dataset_id;
+    get_select_box("datasets", "qc_dataset_select", { "checkbox_name": "qc_dataset_select_checkbox" });
+
+    jQuery('#qc_analysis_prepare_button').removeClass('active').addClass('inactive');
+
+    $(document).on('click', 'input[name=select_engine]', function (e) {
+        get_model_string();
+    });
+
+    $('#qc_analysis_prepare_button').click(function () {
+        dataset_id = get_dataset_id();
+        if (dataset_id != false) {
+            $.ajax({
+                url: '/ajax/qualitycontrol/prepare',
+                data: { 'dataset_id': get_dataset_id() },
```

Comment:
use  dataset_id instead of get_dataset_id()

Reply:
Hey Isaak, It is running on cassava-test correctly. The get_data_id is working correctly there.

Reply:
Yes, I mean you already have stored the output from the get_dataset_id() method in the dataset_id. So, you could just use that...either way, it works...